src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -108,6 +110,7 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 				}
 			}
 		}));
+		this._register(lifecycleService.onWillShutdown(() => this._disposeAllDecorations()));
```

Comment:
Is `onWillShutdown` run when a window is closed or when all windows are closed?

Reply:
there's one `decorationAddon` created for each terminalInstance, so this should be fine 👍🏼 good thought though. works based on my testing